### PR TITLE
The AutoJoinWallsWithColumns works as intended.

### PR DESCRIPTION
**Pull request by samuelpancu94**
It's checking also if the elements are already joined in order to avoid crashing. It makes it more stable.
The AutoJoinWallsWithWalls doesn't work properly. The "secondwall"  in "(JoinGeometryUtils.AreElementsJoined(doc, wall, secondwall)" is null.

### CODE REVIEW

File: Components/AutoJoinWallsWithColumns.cs [hunk 1]
```diff
@@ -46,7 +47,14 @@ public Result Execute(ExternalCommandData commandData, ref string message, Eleme
                     //Join each wall with each column found as intersecting.
                     foreach (Element column in intersectingColumns)
                     {
-                        JoinGeometryUtils.JoinGeometry(doc, column, wall);
+                        if (JoinGeometryUtils.AreElementsJoined(doc, column, wall) == false)
```
Comment (pancudaniel7) [inline Components/AutoJoinWallsWithColumns.cs:50]: Suggestion:
- AreElementsJoined already return a boolean value, you can write code like this: `if (JoinGeometryUtils.AreElementsJoined(doc, column, wall){...`

File: Components/AutoJoinWallsWithColumns.cs [hunk 2]
```diff
@@ -46,7 +47,14 @@ public Result Execute(ExternalCommandData commandData, ref string message, Eleme
...
+                        {
+                            JoinGeometryUtils.JoinGeometry(doc, column, wall);
+                        }
+                        else
+                        {
+                            continue;
+                        }
```
Comment (pancudaniel7) [inline Components/AutoJoinWallsWithColumns.cs:57]: Suggestion:
- You can write this code like this:
`foreach (Element column in intersectingColumns)
{	                    
   if(!JoinGeometryUtils.AreElementsJoined(doc, column, wall)){ break; }
   JoinGeometryUtils.JoinGeometry(doc, column, wall);
}`

File: Components/AutoJoinWallsWithWalls.cs [hunk 3]
```diff
@@ -31,22 +31,60 @@ public Result Execute(ExternalCommandData commandData, ref string message, Eleme
...
+                            if (JoinGeometryUtils.AreElementsJoined(doc, wall, secondwall) == false)
+                            {
+                                TaskDialog.Show("Debugging Test", "Step 10");
+                                JoinGeometryUtils.JoinGeometry(doc, wall, secondwall);
+                                TaskDialog.Show("Debugging Test", "Step 11");
+                            }
+                            else
+                            {
+                                continue;
+                            }
```
Comment (pancudaniel7) [inline Components/AutoJoinWallsWithWalls.cs:66]: Same suggestion here.

Comment (pancudaniel7) [reply Components/AutoJoinWallsWithWalls.cs:66]: Try to avoid to use **continue** keyword

File: Components/AutoJoinWallsWithColumns.cs (hunk 2 above)
Comment (samuelpancu94) [reply Components/AutoJoinWallsWithColumns.cs:57]: Thank you for the tip.

File: Components/AutoJoinWallsWithWalls.cs (hunk 3 above)
Comment (samuelpancu94) [reply Components/AutoJoinWallsWithWalls.cs:66]: Understood, thank you.

**Pull request merged**